bokeh/io.py

```diff
@@ -516,7 +535,7 @@ def push_notebook(document=None, state=None, handle=None):
         msg = Document._compute_patch_between_json(handle.json, to_json)
     handle.comms.send(json.dumps(msg))
     handle._doc = document
-    handle._json = to_json
+    handle._json[id(document)] = to_json
```

Comment:
check out line 532 (just before this diff hunk). it looks like the way it was intended to work is that if `handle.doc is not document` we send the entire document. I don't know why that wouldn't work, unless it's broken on the coffeescript side (doesn't replace the entire document) ? If that worked, we could skip the dict of old json.

I'm a little worried here that we never delete anything from handle._json